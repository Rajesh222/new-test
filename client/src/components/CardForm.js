import React, { Component } from 'react';
import CardList from './CardList';
import {baseUrl} from '../config';

class CardForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            name: '',
            cardNumber: '',
            limit: '',
            balance: '',
            allCardList:[],
            isError: false
        }

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChannge = this.handleChannge.bind(this);
    }

    handleSubmit = (e)=> {
        e.preventDefault();
        const { name, cardNumber, limit, balance } = this.state;
        const data = {
            name, cardNumber, limit, balance
        }
        if (!name && !cardNumber && !limit && !balance) {
            return;
        }
        const options = {
            method: 'POST',
            body: JSON.stringify(data),
            headers: {
                'Content-Type': 'application/json'
            }
        }
        const url = `${baseUrl}/cards`;
        
        fetch(url, options)
            .then(res => res.json())
            .then(res => { 
                this.setState({
                    name: '',
                    cardNumber: '',
                    limit: '',
                    balance: '',
                });
                this.getAllCards();
            }).catch((err) => {
                this.setState({isError: true});
            });
    }
    handleChannge = (e) => {
        let change = {};
        let value = e.target.value
        if (e.target.name === 'cardNumber') {
           value = this.cc_format(value);
        }
        change[e.target.name] = value;
        this.setState(change);
    }
    cc_format = (value)=> {
        let v = value.replace(/\s+/g, '').replace(/[^0-9]/gi, '')
        let matches = v.match(/\d{4,16}/g);
        let match = (matches && matches[0]) || '';
        let parts = [];
        const len =match.length;
        for (let i=0 ; i < len; i+=4) {
          parts.push(match.substring(i, i+4))
        }
        if (parts.length) {
          return parts.join(' ')
        } else {
          return value
        }
    }
    componentDidMount() {
        this.getAllCards();
    }
    getAllCards = () => {
        const url =  `${baseUrl}/cards`;   
        fetch(url)
            .then(res => res.json())
            .then(res => this.setState({allCardList: res.cards}));
    }
    render() {
        const { name, cardNumber, limit, balance, allCardList } = this.state;
        return (
            <div className="card-form">
                <h4>Credit Card System</h4>
                <form onSubmit={this.handleSubmit}>
                    <div className="control-input">Name:<br />
                        <input type="text" name="name" value={name} onChange={this.handleChannge} required/>
                    </div>
                    <div className="control-input">
                        Card number:<br />
                        <input type="text" name="cardNumber" value={cardNumber} onChange={this.handleChannge} required/>
                    </div>
                    <div className="control-input">
                        Limit:<br />
                        <input type="text" name="limit" value={limit} onChange={this.handleChannge} required/>
                    </div>
                    <div className="control-input">
                        Balance:<br />
                        <input type="text" name="balance" value={balance} onChange={this.handleChannge} required/>
                    </div>
                    <div className="control-input">
                        <input className="submit-btn" type="submit" value="Add" />
                    </div>
                </form> 
                <CardList allCardList={allCardList} />
            </div>
        );
    }
}

export default CardForm;
