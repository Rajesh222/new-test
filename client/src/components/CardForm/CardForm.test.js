import React from 'react';
import { mount, shallow } from 'enzyme';
import CardForm from './index';

describe('CardForm component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CardForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('has card-form classname', () => {
    const wrapper = mount(<CardForm />);
    expect(wrapper.find('.card-form').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
