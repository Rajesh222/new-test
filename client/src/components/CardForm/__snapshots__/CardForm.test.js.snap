// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardForm component has card-form classname 1`] = `ReactWrapper {}`;

exports[`CardForm component renders correctly 1`] = `ShallowWrapper {}`;
